docs/questions/visualizations/map.md

```diff
@@ -2,46 +2,195 @@
+
+If you need to display a large number of geographical data points, consider using a grid map to display the distribution instead.
+
+## Grid map
+
+Grid map is an aggregated version of the pin map — like a heatmap for the distribution of pins. Grid map breaks the the map into a grid based on latitude/longitude, and then colors each grid cell based on how many data points fall in it.
+
+![Grid map](../images/grid-map.png)
+
+### Grid map data shape
+
+To create a grid map, you need to have a query returning a metric summarized by binned latitude/longitude coordinates.
+
+![Example of data for grid map](../images/latlong-binned-result.png)
+
+If you're writing a query in the query builder, you can choose how to bin latitude/longitude in the **Group by** block. If you're writing a query in SQL, you'll need to add binning logic yourself.
+
+![Binned latitude and longitude in the query builder](../images/bin-by-latlong.png)
+
+### Create a grid map
+
+To create a grid map:
+
+1. Build a query with summary by binned latitude and longitude columns (either in the query builder or using SQL);
+2. Select **Visualization**, and pick **Map**;
+3. If your query results have columns whose field type is set to latitude/longitude in [table metadata](../../data-modeling/metadata-editing.md#field-type), Metabase should build a grid map automatically.
+
+   Otherwise, click on the **Gear** icon to go to visualization settings, choose **Map type: grid map**, and pick columns that contain latitude and longitude coordinates.
+
+4. If your query contains several metrics, you can pick the one you want to display in visualization settings.
 
 ## Region maps
 
-There are two region maps included with Metabase:
+Region maps display a distribution of aggregated data broken out by regions, like count of users by country.
+
+Metabase comes with two built-in regions: [world map](#world-map) with countries, and the [Unites States map](#united-states-map) with states. Admins can [add custom regions](#custom-regions) in Admin setting.
+
+![Three types of region maps](../images/region-maps.png)
+
+You can disable the default regions by setting the environment variable [`MB_DEFAULT_MAPS_ENABLED`](../../configuring-metabase/environment-variables.md#mb_default_maps_enabled).
+
+### Region map data shape
+
+To build a region map, you need a query (either a question built with the query builder, or a native query) with at least two columns: the column with the region name and the column with the metric that you want to display by region.
+
+![Region map query example](../images/region-map-query.png)
```

Comment:
Adding this image because of numerous questions "but how do I put countries into my queries" even though we already had the note about two letter codes